Framework/API/src/Algorithm.cpp

```diff
@@ -406,15 +405,13 @@ void Algorithm::unlockWorkspaces() {
   if (this->isChild())
     return;
   auto &debugLog = g_log.debug();
-  for (size_t i = 0; i < m_writeLockedWorkspaces.size(); i++) {
-    Workspace_sptr ws = m_writeLockedWorkspaces[i];
+  for (auto ws : m_writeLockedWorkspaces) {
```

Comment:
This comment is about both the old and new code: Why make a copy of the `sptr`? Wouldn't a reference be sufficient and more efficient?